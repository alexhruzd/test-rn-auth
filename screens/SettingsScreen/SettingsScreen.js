import React, { useContext } from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import AuthContext from '../../context/auth/AuthContext';

const SettingsScreen = () => {
  const {onSignOut} = useContext(AuthContext)

  return (
    <View style={styles.container}>
    <Text>Hello!! This is SettingsPage</Text>

    <TouchableOpacity onPress={onSignOut} style={styles.appButtonContainer}>
        <Text style={styles.appButtonText}>Log out</Text>
      </TouchableOpacity>
  </View>
  );
}

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#fff",
  },
  appButtonContainer: {
    elevation: 8,
    backgroundColor: "#009688",
    borderRadius: 10,
    paddingVertical: 10,
    paddingHorizontal: 12,
    width: "50%"
  },
  appButtonText: {
    fontSize: 18,
    color: "#fff",
    fontWeight: "bold",
    alignSelf: "center",
    textTransform: "uppercase",
  },
});

export default SettingsScreen;