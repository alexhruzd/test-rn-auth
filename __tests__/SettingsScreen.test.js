import React from "react";

import { cleanup, render } from "@testing-library/react-native";

import SettingsScreen from "../screens/SettingsScreen";
import AuthContext from "../context/auth/AuthContext";

afterEach(cleanup);

describe("SettingScreen", () => {
  it("should find the button via testId", () => {
    const testIdName = "logOutButton";
    const { toJSON, getByTestId } = render(
      <AuthContext.Provider value={{ onSingOut: () => {} }}>
        <SettingsScreen />
      </AuthContext.Provider>
    );
    const foundButton = getByTestId(testIdName);

    expect(foundButton).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
