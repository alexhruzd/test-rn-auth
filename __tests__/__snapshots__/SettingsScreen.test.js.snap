// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingScreen should find the button via testId 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello!! This is SettingsPage
  </Text>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#009688",
        "borderRadius": 10,
        "elevation": 8,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 10,
        "width": "50%",
      }
    }
    testID="logOutButton"
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      Log out
    </Text>
  </View>
</View>
`;
